varnish.m4

```
@@ -1,104 +1,177 @@
+	AC_REQUIRE([AC_PROG_CC_C99])
+	AC_REQUIRE([AC_PROG_CPP])
+	AC_REQUIRE([AC_PROG_CPP_WERROR])
+
+	AM_PATH_PYTHON([2.6], [], [
+		AC_MSG_ERROR([Python is needed to build VMODs.])
+	])
+
+	AS_IF([test -z "$RST2MAN"], [
+		AC_MSG_ERROR([rst2man is needed to build VMOD manuals.])
+	])
+
+	dnl Expose the location of the std and directors VMODs
+	AC_SUBST([VARNISHAPI_VMODDIR])
+
+	dnl Expose Varnish's aclocal directory to automake
+	AC_SUBST([VARNISHAPI_DATAROOTDIR])
+
+	dnl Define the VMOD directory for libtool
+	AS_CASE([$prefix],
+		[NONE], [
+			vmoddir=$VARNISHAPI_VMODDIR
+			ac_default_prefix=$VARNISHAPI_PREFIX],
+		[vmoddir=$libdir/varnish/vmods]
+	)
+	AC_SUBST([vmoddir])
+
+	dnl Define an automake silent execution for vmodtool
+	[am__v_VMODTOOL_0='@echo "  VMODTOOL" $<;']
+	[am__v_VMODTOOL_1='']
+	[am__v_VMODTOOL_='$(am__v_VMODTOOL_$(AM_DEFAULT_VERBOSITY))']
+	[AM_V_VMODTOOL='$(am__v_VMODTOOL_$(V))']
+	AC_SUBST([am__v_VMODTOOL_0])
+	AC_SUBST([am__v_VMODTOOL_1])
+	AC_SUBST([am__v_VMODTOOL_])
+	AC_SUBST([AM_V_VMODTOOL])
+
+	dnl Define VMODs LDFLAGS
+	AC_SUBST([VMOD_LDFLAGS],
+		"-module -export-dynamic -avoid-version -shared")
+
+	dnl Define the PATH for the test suite
+	AC_SUBST([VMOD_TEST_PATH],
+		[$VARNISHAPI_SBINDIR:$VARNISHAPI_BINDIR:$PATH])
+])
+
+# _VARNISH_VMOD(NAME)
+# -------------------
+AC_DEFUN([_VARNISH_VMOD], [
+
+	AC_REQUIRE([_VARNISH_VMOD_CONFIG])
+
+	VMOD_FILE="\$(abs_builddir)/.libs/libvmod_$1.so"
+	AC_SUBST(m4_toupper(VMOD_$1_FILE), [$VMOD_FILE])
+
+	VMOD_IMPORT="querystring from \\\"$VMOD_FILE\\\""
+	AC_SUBST(m4_toupper(VMOD_$1), [$VMOD_IMPORT])
+
+	VMOD_RULES="
+
+vmod_$1.lo: vcc_$1_if.c vcc_$1_if.h
+
+vcc_$1_if.c vcc_$1_if.h vmod_$1.rst vmod_$1.man.rst: .vcc_$1
+
+.vcc_$1: vmod_$1.vcc
+	\$(AM_V_VMODTOOL) $PYTHON $VMODTOOL -o vcc_$1_if \$(srcdir)/vmod_$1.vcc
+	@touch .vcc_$1
+
+vmod_$1.3: vmod_$1.man.rst
+	$RST2MAN vmod_$1.man.rst vmod_$1.3
+
+clean: clean-vmod-$1
+
+distclean: clean-vmod-$1
+
+clean-vmod-$1:
+	rm -f vcc_$1_if.c vcc_$1_if.h
```

should ignore errors here (prefix lines with -)